homeassistant/components/geonetnz_quakes/__init__.py

```diff
@@ -99,6 +129,116 @@
+        )
+        websession = aiohttp_client.async_get_clientsession(hass)
+        self._feed_manager = GeonetnzQuakesFeedManager(
+            websession,
+            self._generate_entity,
+            self._update_entity,
+            self._remove_entity,
+            coordinates,
+            mmi=config_entry.data[CONF_MMI],
+            filter_radius=radius_in_km,
+            filter_minimum_magnitude=config_entry.data[CONF_MINIMUM_MAGNITUDE],
+            filter_time=DEFAULT_FILTER_TIME_INTERVAL,
+            status_callback=self._status_update,
+        )
+        self._config_entry_id = config_entry.entry_id
+        self._scan_interval = timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL])
+        self._unit_system = unit_system
+        self._track_time_remove_callback = None
+        self._status_info = None
+        self.listeners = []
+
+    async def async_init(self):
+        """Schedule initial and regular updates based on configured time interval."""
+
+        for domain in COMPONENTS:
+            self._hass.async_create_task(
+                self._hass.config_entries.async_forward_entry_setup(
+                    self._config_entry, domain
+                )
+            )
+
+        async def update(event_time):
+            """Update."""
+            await self.async_update()
+
+        self._track_time_remove_callback = async_track_time_interval(
+            self._hass, update, self._scan_interval
+        )
+
+        async def component_loaded(event):
+            """Handle a new component loaded."""
+            component = event.data.get(ATTR_COMPONENT)
+            if component == DOMAIN:
+                await self.async_update()
+
+        self._hass.bus.async_listen(EVENT_COMPONENT_LOADED, component_loaded)
+
+        _LOGGER.debug("Feed entity manager initialized")
+
+    async def async_update(self):
+        """Refresh data."""
+        await self._feed_manager.update()
+        _LOGGER.debug("Feed entity manager updated")
+
+    async def async_stop(self):
+        """Stop this feed entity manager from refreshing."""
+        for unsub_dispatcher in self.listeners:
+            unsub_dispatcher()
+        self.listeners = []
+        if self._track_time_remove_callback:
+            self._track_time_remove_callback()
+        _LOGGER.debug("Feed entity manager stopped")
+
+    @callback
+    def async_event_new_entity(self):
+        """Return manager specific event to signal new entity."""
+        return SIGNAL_NEW_GEOLOCATION.format(self._config_entry_id)
+
+    def get_entry(self, external_id):
+        """Get feed entry by external id."""
+        return self._feed_manager.feed_entries.get(external_id)
+
+    def status_info(self):
+        """Return latest status update info received."""
+        return self._status_info
+
+    async def _generate_entity(self, external_id):
+        """Generate new entity."""
+        new_entity = GeonetnzQuakesEvent(self, external_id, self._unit_system)
```

Comment:
The entity should be created in the platform. Use the dispatch helper to signal a function that does this.

Reply:
OK. I moved the actual creation of the entity to the platform.